continuationsPlugin/src/test/scala/continuations/CompilerFixtures.scala

```diff
@@ -174,25 +194,298 @@ trait CompilerFixtures { self: FunSuite =>
+                    Block(
+                      Nil,
+                      ref(continuationVal).select(Names.termName("resume")).appliedTo(rightOne))
+                  )),
+                  Closure(Nil, ref(anonFunc), TypeTree(c.definitions.UnitType))
+                )
+              ))
+          ),
+          List(ref(usingSuspend))
+        )
+
+      DefDef(
+        Symbols.newSymbol(c.owner, Names.termName("mySuspend"), EmptyFlags, intType).asTerm,
+        List(List(), List(usingSuspend)),
+        intType,
+        rhs
+      )
+    },
+    teardown = _ => ()
+  )
+
+  val zeroAritySuspendNonSuspendingDefDef = FunFixture[Context ?=> DefDef](
+    setup = _ => {
+      val c = summon[Context]
+      val suspend = Symbols.requiredClass("continuations.Suspend")
+      val right = Symbols.requiredModule("scala.util.Right")
+
+      val intType = c.definitions.IntType
+
+      val usingSuspend =
+        Symbols.newSymbol(
+          c.owner,
+          Names.termName("x$1"),
+          Flags.union(Flags.GivenOrImplicit, Flags.Param),
+          suspend.info)
+
+      val rhs =
+        Literal(Constant(1))
+
+      DefDef(
+        Symbols.newSymbol(c.owner, Names.termName("mySuspend"), EmptyFlags, intType).asTerm,
+        List(List(), List(usingSuspend)),
+        intType,
+        rhs
+      )
+    },
+    teardown = _ => ()
+  )
+
+  val continuationsContextAndZeroArityNonSuspendNonSuspendingDefDef =
+    FunFixture.map2(
+      compilerContextWithContinuationsPlugin,
+      zeroArityNonSuspendNonSuspendingDefDef)
+
+  val continuationsContextAndZeroAritySuspendSuspendingDefDef =
+    FunFixture.map2(compilerContextWithContinuationsPlugin, zeroAritySuspendSuspendingDefDef)
+
+  val continuationsContextAndZeroAritySuspendNonSuspendingDefDef =
+    FunFixture.map2(compilerContextWithContinuationsPlugin, zeroAritySuspendNonSuspendingDefDef)
+
+  val continuationsContextAndSuspendContextFunction =
+    FunFixture.map2(compilerContextWithContinuationsPlugin, suspendContextFunction)
+
+  val continuationsContextAndNonSuspendContextFunctionReturingSuspend = FunFixture.map2(
+    compilerContextWithContinuationsPlugin,
+    nonSuspendContextFunctionReturingSuspend)
+
+  val continuationsContextAndSuspendContextFunctionReturningSuspend = FunFixture.map2(
+    compilerContextWithContinuationsPlugin,
+    suspendContextFunctionReturningSuspend)
+
+  val continuationsContextAndNonSuspendContextFunction = FunFixture.map2(
+    compilerContextWithContinuationsPlugin,
+    nonSuspendContextFunction
+  )
+
+  val continuationsContextAndZeroArityContextFunctionDefDef = FunFixture.map2(
+    compilerContextWithContinuationsPlugin,
+    zeroArityContextFunctionDefDef
+  )
+
+  val continuationsContextAndSuspendingContextFunctionValDef = FunFixture.map2(
+    compilerContextWithContinuationsPlugin,
+    suspendingContextFunctionValDef
+  )
+
+  val continuationsContextAndNonSuspendingContextFunctionValDef = FunFixture.map2(
```

Comment:
nice, these fixtures make the code nicer and easier to read for now. and definitely easier to write more tests as transformations are being added

Reply:
😆 I think we may need a better naming scheme. The arity + plugin + context gets kind of wordy. Ideas?

Reply:
ha yes I didn't say anything about that because of the absence of ideas. maybe something like `nonSuspendingVal` and having it in a `object ContinuationFixtures{...}` would also make sense?

naming is hard...